pkg/ccl/backupccl/restore_online.go

```diff
@@ -501,14 +503,35 @@ func (r *restoreResumer) cleanupAfterDownload(
 ) error {
 	ctx, sp := tracing.ChildSpan(ctx, "backupccl.cleanupAfterDownload")
 	defer sp.Finish()
+	if len(details.TableDescs) == 0 {
+		panic("no table descs")
+	}
 
-	executor := r.execCfg.InternalDB.Executor()
-
-	// Re-enable automatic stats collection on restored tables.
+	// Try to restore automatic stats collection preference on each restored
+	// table.
 	for _, table := range details.TableDescs {
-		_, err := executor.Exec(ctx, "enable-stats", nil, `ALTER TABLE $1 SET (sql_stats_automatic_collection_enabled = true);`, table.Name)
-		if err != nil {
-			log.Warningf(ctx, "could not enable automatic stats on table %s", table)
+		if err := sql.DescsTxn(ctx, r.execCfg, func(
+			ctx context.Context, txn isql.Txn, descsCol *descs.Collection,
+		) error {
+			b := txn.KV().NewBatch()
+			newTableDesc, err := descsCol.MutableByID(txn.KV()).Table(ctx, table.ID)
+			if err != nil {
+				return err
+			}
+			newTableDesc.AutoStatsSettings = &catpb.AutoStatsSettings{}
```

Comment:
one sad part about writing the download job after publishing descriptors is that `tableDescs` contain the in-memory representation of the _published_ descriptors, so `tableDescs[0].AutoStatsSettings` will always be `false`. If we grabbed the `tableDescs` before publishing the descriptors, then `tableDescs[0].AutoStatsSettings` would have been set to whatever was in the backup. To get around this, we could make a deep copy of `tableDescs` before publishing the descriptors and pass that to `maybeWriteDownloadJob` . I don't really like this idea either.

Reply:
🤔 Copying aside for a moment:

I wonder if it would be worth asking foundations what they would think about an additional field in the stats settings that was something like `DisabledDuringRestore` or something.  We would then change some code in `func (as *AutoStatsSettings) AutoStatsCollectionEnabled() AutoStatsCollectionStatus` to additionally look at that field.  One reason it might be nice is that it could be used to drive other UX around what happens if the user changes the setting via SQL (since the tables will be online).

Reply:
This is a great idea. let me ping sql queries.

Reply:
Decided to stick with the approach of keeping all logic in backupccl.